src/main/org/codehaus/groovy/reflection/ClassInfo.java

```diff
@@ -35,13 +36,13 @@
  *
  * @author Alex.Tkachman
  */
-public class ClassInfo {
+public class ClassInfo implements Finalizable {
 
     private final LazyCachedClassRef cachedClassRef;
     private final LazyClassLoaderRef artifactClassLoader;
     private final LockableObject lock = new LockableObject();
     public final int hash = -1;
-    private final Class klazz;
+    private final WeakReference<Class<?>> klazz;
```

Comment:
That's the question I've struggled with too.  My assumption here is that only the Groovy Runtime should ever ask for ClassInfo objects and in most places I can find they do so either with a Class or an instance of the Class which I think would mean it wouldn't have been collected.  But it's been difficult for me to be certain that the case can never happen.  Best I can think of is maybe check if the ref is null and throw a `GroovyBugError`.